// mnemonist/index.js
/**
 * Mnemonist Library Endpoint (CommonJS)
 * ======================================
 *
 * Exporting every data structure through a unified endpoint. Consumers
 * of this library should prefer the modular access though.
 */
var Heap = require('mnemonist/heap.js'),
    FibonacciHeap = require('mnemonist/fibonacci-heap.js'),
    SuffixArray = require('mnemonist/suffix-array.js'),
    Vector = require('mnemonist/vector.js');

module.exports = {
  BiMap: require('mnemonist/bi-map.js'),
  BitSet: require('mnemonist/bit-set.js'),
  BitVector: require('mnemonist/bit-vector.js'),
  BloomFilter: require('mnemonist/bloom-filter.js'),
  BKTree: require('mnemonist/bk-tree.js'),
  CircularBuffer: require('mnemonist/circular-buffer.js'),
  DefaultMap: require('mnemonist/default-map.js'),
  DefaultWeakMap: require('mnemonist/default-weak-map.js'),
  FixedDeque: require('mnemonist/fixed-deque.js'),
  StaticDisjointSet: require('mnemonist/static-disjoint-set.js'),
  FibonacciHeap: FibonacciHeap,
  MinFibonacciHeap: FibonacciHeap.MinFibonacciHeap,
  MaxFibonacciHeap: FibonacciHeap.MaxFibonacciHeap,
  FixedReverseHeap: require('mnemonist/fixed-reverse-heap.js'),
  FuzzyMap: require('mnemonist/fuzzy-map.js'),
  FuzzyMultiMap: require('mnemonist/fuzzy-multi-map.js'),
  HashedArrayTree: require('mnemonist/hashed-array-tree.js'),
  Heap: Heap,
  MinHeap: Heap.MinHeap,
  MaxHeap: Heap.MaxHeap,
  StaticIntervalTree: require('mnemonist/static-interval-tree.js'),
  InvertedIndex: require('mnemonist/inverted-index.js'),
  KDTree: require('mnemonist/kd-tree.js'),
  LinkedList: require('mnemonist/linked-list.js'),
  LRUCache: require('mnemonist/lru-cache.js'),
  LRUCacheWithDelete: require('mnemonist/lru-cache-with-delete.js'),
  LRUMap: require('mnemonist/lru-map.js'),
  LRUMapWithDelete: require('mnemonist/lru-map-with-delete.js'),
  MultiMap: require('mnemonist/multi-map.js'),
  MultiSet: require('mnemonist/multi-set.js'),
  PassjoinIndex: require('mnemonist/passjoin-index.js'),
  Queue: require('mnemonist/queue.js'),
  FixedStack: require('mnemonist/fixed-stack.js'),
  Stack: require('mnemonist/stack.js'),
  SuffixArray: SuffixArray,
  GeneralizedSuffixArray: SuffixArray.GeneralizedSuffixArray,
  set: require('mnemonist/set.js'),
  SparseQueueSet: require('mnemonist/sparse-queue-set.js'),
  SparseMap: require('mnemonist/sparse-map.js'),
  SparseSet: require('mnemonist/sparse-set.js'),
  SymSpell: require('mnemonist/symspell.js'),
  Trie: require('mnemonist/trie.js'),
  TrieMap: require('mnemonist/trie-map.js'),
  Vector: Vector,
  Uint8Vector: Vector.Uint8Vector,
  Uint8ClampedVector: Vector.Uint8ClampedVector,
  Int8Vector: Vector.Int8Vector,
  Uint16Vector: Vector.Uint16Vector,
  Int16Vector: Vector.Int16Vector,
  Uint32Vector: Vector.Uint32Vector,
  Int32Vector: Vector.Int32Vector,
  Float32Vector: Vector.Float32Vector,
  Float64Vector: Vector.Float64Vector,
  PointerVector: Vector.PointerVector,
  VPTree: require('mnemonist/vp-tree.js')
};
